ecl/hql/hqlutil.cpp

```diff
@@ -8285,8 +8286,21 @@ void extractXmlName(StringBuffer & name, StringBuffer * itemName, StringBuffer *
 
         unsigned lenContents = strlen(XPATH_CONTENTS_TEXT);
         unsigned lenTagName = tagName.length();
-        if ((lenTagName >= lenContents) && (memcmp(tagName.str() + (lenTagName - lenContents), XPATH_CONTENTS_TEXT, lenContents) == 0))
-            tagName.setLength(lenTagName - lenContents);
+        if (lenTagName >= lenContents)
+        {
+            const char *finger = tagName.str() + (lenTagName - lenContents);
+            if (memcmp(finger, XPATH_CONTENTS_TEXT, lenContents) == 0)
+            {
+                bool isNamed = ((lenTagName >= lenContents + 1) && *--finger=='/');
```

Comment:
minor: finger[-1] cleaner than *--finger, since the latter implies you are going to use the updated variable later on.